src/snapred/backend/recipe/algorithm/LoadGroupingDefinition.py

```diff
@@ -111,6 +111,7 @@ def PyExec(self) -> None:
                 instrument_donor = self.mantidSnapper.LoadEmptyInstrument(
                     "Loading instrument definition file...",
                     Filename=self.getProperty("InstrumentFilename").value,
+                    InstrumentName=self.getProperty("InstrumentName").value,
```

Comment:
Here you can actually use all of `InstrumentName`, `InstrumentDonor`, and `Filename` and `LoadEmptyInstrument` should handle it

Reply:
Are you sure I need InstrumentDonor? That's what this section is setting, and it only gets here if InstrumentDonor isn't set. I will add InstrumentFilename though

Reply:
You're right, I'm wrong.